### PR TITLE
chore: update containerd version to v1.7.9 for 23H2

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind update

**What this PR does / why we need it**:
Update containerd version to v1.7.9 for 23H2.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/windows/windowscsehelper.ps1 [hunk 1]
```diff
@@ -302,6 +305,11 @@ function Install-Containerd-Based-On-Kubernetes-Version {
     if (([version]$KubernetesVersion).CompareTo([version]$global:MinimalKubernetesVersionWithLatestContainerd) -ge 0) {
       $containerdPackage=$global:LatestContainerdPackage
       Write-Log "Kubernetes version $KubernetesVersion is greater than or equal to $global:MinimalKubernetesVersionWithLatestContainerd so the latest containerd version $containerdPackage is used"
+      $buildNumber = (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").CurrentBuild
```
Comment (AbelHu) [inline parts/windows/windowscsehelper.ps1:308]: 1. Suggest to add a function to get BuildNumber and reuse it in Get-WindowsVersion.
2. Update ut

File: parts/windows/windowscsehelper.ps1 [hunk 2]
```diff
@@ -302,6 +309,11 @@ function Install-Containerd-Based-On-Kubernetes-Version {
     if (([version]$KubernetesVersion).CompareTo([version]$global:MinimalKubernetesVersionWithLatestContainerd) -ge 0) {
       $containerdPackage=$global:LatestContainerdPackage
```
Comment (AbelHu) [inline parts/windows/windowscsehelper.ps1:310]: We will have two logs for 23H2 so I think that it is better to use else for setting $global:LatestContainerdPackage

Comment (junjiezhang1997) [reply parts/windows/windowscsehelper.ps1]: Fixed.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**